index.bs

```diff
@@ -1557,6 +1685,86 @@ The [=remote end steps=] are:
 
 2. Return [=success=] with data |body|
 
+#### The session.new Command #### {#command-session-new}
+
+The <dfn export for=commands>session.new</dfn> command allows creating a new
+[=WebDriver BiDi session=].
+
+Note: A session created this way will not be accessible via HTTP.
+
+This is a [=static command=].
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      SessionNewCommand = {
+        method: "session.new",
+        params: {capabilities: CapabilitiesRequestParameters},
+      }
+
+      CapabilitiesRequestParameters = {
+        ?alwaysMatch: CapabilitiesRequest,
+        ?firstMatch: [*CapabilitiesRequest]
+      }
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+      SessionNewResult = {
+        sessionId: text,
+        capabilities: {
```

Comment:
Some of these make sense in the context of WebDriver over HTTP, but not really for BiDi if connecting directly to a browser. `browserName` in particular :)

I'm also not sure about whether it might be too late for `acceptInsecureCertificates` and `proxy`, if those might be things that are implemented as command line arguments.

Are there any issues for Firefox to implement this as written?

Reply:
Well for a direct connection to a browser `browserName` doesn't make any sense over HTTP either. The use case is something like Selenium Grid, and it at least seems plausible that the same thing can work without requiring a HTTP session: the grid server provides a WebSocket endpoint to connect to, and handles the browser selection and setup.

For for `acceptInsecureCertificates` and `proxy`, if you can't set those at runtime, and the browser is already started, you'd reject the session. But in a case like Puppeteer where the same software starts the browser and then generates a new session request that won't happen. It does means duplicating some information between the command line arguments (or profile settings, or whatever), but it also allows the same setup to work when the software that starts the browser isn't the one generating the new session message (again consider the case of a setup with middleware like a grid server; in this case the client doesn't start the browser but connects to the server and sends it the new session message, which it uses to configure the browser and then forwards on).

For Firefox in particular I think we can set those things at runtime and not just at startup time. But of course there are other things (like command line arguments) which we can't set at runtime.

Reply:
I see, the behavior when connecting directly to a browser is defined then and not a mystery. Some of this might be tricky to implement when the protocol is built into the browser binary and we might not notice until it's built in, but anyway the model seems sensible.